app/models/puppetclass.rb

```diff
@@ -45,30 +46,133 @@ class Puppetclass < ActiveRecord::Base
 
   scope :not_in_any_environment, lambda { includes(:environment_classes).where(:environment_classes => {:environment_id => nil}) }
 
-  # returns a hash containing modules and associated classes
-  def self.classes2hash(classes)
-    hash = {}
-    for klass in classes
-      if (mod = klass.module_name)
-        hash[mod] ||= []
-        hash[mod] << klass
-      else
-        next
+  class << self
+    # returns a hash containing modules and associated classes
+    def classes2hash(classes)
+      hash = {}
+      for klass in classes
+        if (mod = klass.module_name)
+          hash[mod] ||= []
+          hash[mod] << klass
+        else
+          next
+        end
+      end
+      hash
+    end
+
+    # For API v2 - eliminate node :puppetclass for each object. returns a hash containing modules and associated classes
+    def classes2hash_v2(classes)
+      hash = {}
+      classes.each do |klass|
+        if (mod = klass.module_name)
+          hash[mod] ||= []
+          hash[mod] << {:id => klass.id, :name => klass.name, :created_at => klass.created_at, :updated_at => klass.updated_at}
+        end
+      end
+      hash
+    end
+
+    # Populates the rdoc tree with information about all the classes in your modules.
+    #   Firstly, we prepare the modules tree
+    #   Secondly we run puppetdoc over the modulespath and manifestdir for all environments
+    # The results are written into document_root/puppet/rdoc/<env>/<class>"
+    def self.rdoc(root)
```

Comment:
without self?
overall, maybe should be in a different commit if its only change self